File: app/src/main/java/com/example/gastosinteligentes/database/entidades/Presupuesto.java
package com.example.gastosinteligentes.database.entidades;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(
        tableName = "presupuestos",
        foreignKeys = @ForeignKey(
                entity = Usuario.class,
                parentColumns = "id",
                childColumns = "id_usuario",
                onDelete = ForeignKey.CASCADE
        )
)
public class Presupuesto {

    @PrimaryKey(autoGenerate = true)
    public int id;

    public double monto;
    public int mes;
    public int anio;

    public int id_usuario;
}

File: app/src/main/java/com/example/gastosinteligentes/database/entidades/Presupuesto.kt
package com.example.gastosinteligentes.database.entidades

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "presupuestos",
    foreignKeys = [
        ForeignKey(
            entity = Usuario::class,
            parentColumns = ["id"],
            childColumns = ["id_usuario"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Presupuesto(

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0,

    var monto: Double = 0.0,

    var mes: Int = 0,

    var anio: Int = 0,

    var id_usuario: Int = 0
)